specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/Product.json

```diff
@@ -143,6 +143,147 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/_all/GetProducts": {
+      "post": {
+        "x-ms-examples": {
+          "Returns a list of products.": {
+            "$ref": "examples/Product/ListPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns a list of products.",
+        "operationId": "GetProducts_List",
```

Comment:
rename to Products_GetProducts ? sounds awful but we have the api like that!!

Reply:
If I rename it to Products_GetProducts - i receive a couple more warnings by "autorest" tool. It actually forces this name if I want to avoid warnings.

I still can rename and live with the warnings, though, Should I?

Reply:
The convention is `Noun_Verb` where `Noun` is plural.  Since this returns a list of products I think the name should be `Products_List`

Reply:
Again. If I rename it, I get autorest warning:
WARNING (PostOperationIdContainsUrlVerb/R2066/SDKViolation): OperationId should contain the verb: 'getproducts' in:'Products_List'. Consider updating the operationId

Should I ignore it and change the name?

Reply:
We can leave it as `Products_GetProducts`, it just means that for method group `Products` (i.e. IProducts) there will be two methods, GetProduct and GetProducts.

Reply:
renamed to Products_GetProducts